content/TypeHinting/TypeHinting.q.md

```diff
@@ -0,0 +1,1346 @@
+
+
+/// type=SS, id=23d532c0-d8ac-4efa-824c-10779ecba445, answer=[2]
+
+On line 2, what does `declare(strict_types = 1);` do?
+
+ - It sets the value `1` as a default type declaration.
+
+ - It enables strict type mode validations within the file.
+
+ - It creates the `declare()` method of the `Person` class.
+
+ - It declares the `strict_types` property of the `Person` class.
+
+ - It applies type declarations to methods of the `Person` class.
+
+
+/// type=SS, id=a8a1a5cf-c755-4a2e-94b4-44badc82e9c3, answer=[3]
+
+In the method declaration `public function setName(string $name){}` on line 8, what is `string`?
+
+ - It is a keyword.
+
+ - It is a directive.
+
+ - It is a data type.
+
+ - It is a construct.
+
+ - It is an operator.
+
+
+/// type=SS, id=9084ed22-2c00-4635-abc5-1dac7491cecb, answer=[2]
+
+In the method declaration `public function setName(string $name){}` on line 8, what is `$name`?
+
+ - It is a keyword.
+
+ - It is a parameter.
+
+ - It is a data type.
+
+ - It is a construct.
+
+ - It is an operator.
+
+
+/// type=MS, id=99b8cd12-60c5-40ea-88ab-928efb083348, answer=[3,5]
+
+In the method declaration `public function setName(string $name){}` on line 8, what does `string $name` do?
+
+ - It calls the `$name` property of the `Person` class.
+
+ - It defines the `$name` property of the `Person` class.
+
+ - It sets the data type of the `$name` parameter to `string`.
+
+ - It creates the `$name` parameter of the `setName()` method.
+
+ - It enables the `setName()` method to only accept values of type `string` as an argument.
+
+
+/// type=MS, id=2ef9b03a-5b3f-4e82-89cf-3b94b8e81ce5, answer=[2,5]
+
+Which statements correctly describe the `setName()` method of the `Person` class?
+
+ - It is accessible only within the `Person` class.
+
+ - It only accepts a value of type `string` as an argument.
+
+ - It only returns a value of type `string` when it is called.
+
+ - It is not accessible by any instance of the `Person` class.
+
+ - It sets a value of the `$name` property of the `Person` class.
+
+
+/// type=MS, id=5493b532-c275-4943-8e74-36e6932c0106, answer=[1,3]
+
+Which statements correctly describe the `getName()` method of the `Person` class?
+
+ - It does not require an argument.
+
+ - It only accepts a value of type `string` as an argument.
+
+ - It only returns a value of type `string` when it is called.
+
+ - It is not accessible by any instance of the `Person` class.
+
+ - It sets a value of the `$name` property of the `Person` class.
+
+:::
+
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+declare(strict_types = 1);
+
+class Person 
+{
+    protected $name;
+
+    public function __construct(string $name)
+    {
+        $this->name = $name;
+    }
+
+    public function setName(string $name)
+    {
+        $this->name = $name;
+    }
+
+    public function getName() : string
+    {
+        return $this->name;
+    }
+}
+
+class Car
+{
+    protected $driver;
+
+    public function __construct(Person $driver)
```

Comment:
@rosoarv Object type hinting is already included in the content. This is the example of Object type hint... Please check.

Reply:
@rosoarv I will add another REPL and questions for Arrays but for callable I think we need to add another lesson for that since we are not able to discuss callable in our PHP courses. Thank you.